src/test/ui/infinite/infinite-vec-type-recursion.stderr

```diff
@@ -1,10 +1,10 @@
-error[E0391]: cycle detected when computing type of `X`
+error[E0391]: cycle detected when expanding type alias `X`
   --> $DIR/infinite-vec-type-recursion.rs:1:14
    |
 LL | type X = Vec<X>;
    |              ^
    |
-   = note: ...which again requires computing type of `X`, completing the cycle
+   = note: ...which immediately again requires expanding type alias `X`
```

Comment:
It'd be really nice to add a note that says something like "type aliases cannot be recursive" or "type aliases cannot reference themselves", but I'm not sure how to do that in a non-hacky way.

Reply:
And to also add a hint to use a struct or enum as #17539 suggests.

Reply:
I think that the least hacky way of accomplishing that would be to make a `DefId` part of the `QueryStackFrame`. I don't know what the perf implications might be of doing so, but that will be the most foolproof way of approaching this. Alternatively I can think of making a substring search for `type alias`, but I don't condone such things except as last resort, even though I've been guilty of them in the past 😅.

Reply:
Would it be better to just store the `DefKind`? I think that could work too.

Reply:
I think it should work. Can you try it out?